sympy/functions/elementary/trigonometric.py

```diff
@@ -1805,6 +1843,11 @@ def fdiff(self, argindex=1):
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_is_complex(self):
+        arg = self.args[0]
+        if arg.is_real and (arg / pi).is_integer is False:
+            return True
+
     @staticmethod
     @cacheit
```

Comment:
same function with same name and signature as in line 1767 with different implementation. However, it is in different classes. Why not keep one version of the function. The most efficient version of the function. Or move to utilities folder as it is a common module to prevent duplication

Reply:
I don't understand. We have two difference classes `sec` and `csc` with two different methods:
https://github.com/sympy/sympy/blob/2710561e39c93c01e930173d5b9301e2ae69703e/sympy/functions/elementary/trigonometric.py#L1767-L1776
https://github.com/sympy/sympy/blob/2710561e39c93c01e930173d5b9301e2ae69703e/sympy/functions/elementary/trigonometric.py#L1853-L1863
What's wrong with that?